core/controllers/profile.py

```diff
@@ -309,6 +425,44 @@ def get(self):
 class SignupHandler(base.BaseHandler):
     """Provides data for the editor prerequisites page."""
 
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'POST': {
+            'username': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_valid_username_string'
+                    }]
+                },
+                'default_value': None
+            },
+            'agreed_to_terms': {
+                'schema': {
+                    'type': 'bool'
+                },
+                'default_value': False
+            },
+            'default_dashboard': {
+                'schema': {
+                    'type': 'basestring',
+                    'choices': [
+                        constants.DASHBOARD_TYPE_LEARNER,
+                        constants.DASHBOARD_TYPE_CREATOR
+                    ]
+                },
+                'default_value': None
```

Comment:
Why `None`?

Reply:
From a few requests, I also found it as an optional arg. That's why I added none as default_value. Please tell me if I did wrong.

Reply:
Which requests?

Reply:
I mean after running the server, I hit the handler from the frontend by giving some data as input and I meant it as a request.

Reply:
Can you link the frontend code that does this request?

Reply:
Same as below.